ark/schema/shared/errors.ts

```diff
@@ -279,6 +287,24 @@ export class ArkErrors
 	}
 }
 
+export class TraversalError extends Error {
+	name = "TraversalError"
+	arkErrors!: ArkErrors
+
+	constructor(errors: ArkErrors) {
+		if (errors.length === 1) super(errors.summary)
+		else super("\n" + errors.map(error => `  • ${indent(error)}`).join("\n"))
+
+		Object.defineProperty(this, "arkErrors", {
+			value: errors,
+			enumerable: false
+		})
+	}
+}
+
+const indent = (error: ArkError): string =>
+	error.toString().split("\n").join("\n  ")
```

Comment:
Maybe more efficient to do a replace here?

Reply:
Good catch, I used to put a `map` in between but now it's redundant to do both.